src/abstractdatatable/join.jl

```diff
@@ -3,217 +3,170 @@
 ##
 
 # Like similar, but returns a nullable array
-similar_nullable{T}(dv::AbstractArray{T}, dims::@compat(Union{Int, Tuple{Vararg{Int}}})) =
```

Comment:
Why remove these? Either the function isn't used at all now, or all methods should be kept.